hook.php

```diff
@@ -341,12 +341,42 @@ function plugin_fields_addWhere($link, $nott, $itemtype, $ID, $val, $searchtype)
     /** @var \DBmysql $DB */
     global $DB;
 
-    $searchopt = &Search::getOptions($itemtype);
-    $table     = $searchopt[$ID]['table'];
-    $field     = $searchopt[$ID]['field'];
+    $searchopt    = &Search::getOptions($itemtype);
+    $table        = $searchopt[$ID]['table'];
+    $field        = $searchopt[$ID]['field'];
+    $pfields_type = $searchopt[$ID]['pfields_type'] ?? '';
 
     $field_field = new PluginFieldsField();
 
+    if (
+        $field_field->getFromDBByCrit(
+            [
+                'name'     => $field,
+                'type' => 'number',
+            ],
+        )
+        && $pfields_type == 'number'
+    ) {
+        // if 'number' field with name is found with searchtype 'equals' or 'notequals'
+        // update WHERE clause with `$table_$field.$field` because without `$table_$field.id` is used
+        if ($searchtype == 'equals' || $searchtype == 'notequals') {
+            $operator = ($searchtype == 'equals') ? '=' : '!=';
+            if ($nott) {
+                $link = $link . ' NOT ';
+            }
+            return $link . 'CAST(' . $DB->quoteName("$table" . '_' . "$field") . '.' . $DB->quoteName($field) . ' AS DECIMAL(10,2))' . $operator . ' ' . $DB->quoteValue($val) ;
```

Comment:
```suggestion
            return $link . 'CAST(' . $DB->quoteName("$table" . '_' . "$field") . '.' . $DB->quoteName($field) . ' AS DECIMAL(10,9))' . $operator . ' ' . $DB->quoteValue($val) ;
```

In case when user have saved decimal like 10.2568 (with more than two digit after comma)